src/network/network_connection.cpp

```diff
@@ -21,53 +26,80 @@ void NetworkConnection::Init(short event_flags, NetworkThread *thread,
   SetNonBlocking(sock_fd);
   SetTCPNoDelay(sock_fd);
 
+  protocol_handler_ = nullptr;
+
   this->event_flags = event_flags;
   this->thread = thread;
   this->state = init_state;
 
   this->thread_id = thread->GetThreadID();
 
   // clear out packet
-  rpkt.Reset();
-  if (event == nullptr) {
-    event = event_new(thread->GetEventBase(), sock_fd, event_flags,
+  if (network_event == nullptr) {
+    network_event = event_new(thread->GetEventBase(), sock_fd, event_flags,
                       CallbackUtil::EventHandler, this);
   } else {
     // Reuse the event object if already initialized
-    if (event_del(event) == -1) {
-      LOG_ERROR("Failed to delete event");
+    if (event_del(network_event) == -1) {
+      LOG_ERROR("Failed to delete network event");
       PL_ASSERT(false);
     }
 
-    auto result = event_assign(event, thread->GetEventBase(), sock_fd,
+    auto result = event_assign(network_event, thread->GetEventBase(), sock_fd,
                                event_flags,
                                CallbackUtil::EventHandler, this);
 
     if (result != 0) {
-      LOG_ERROR("Failed to update event");
+      LOG_ERROR("Failed to update network event");
+      PL_ASSERT(false);
+    }
+  }
+
+  if (workpool_event == nullptr) {
+    workpool_event = event_new(thread->GetEventBase(), -1, EV_WRITE|EV_PERSIST,
```

Comment:
Minor thing: why do we set the EV_WRITE flag if fd=-1 (the event needs to be triggered manually)?

Reply:
Yes, when the worker thread in worker pool finishes execution, it uses event_active to manually trigger the network thread. This event is only triggered manually.

Reply:
I know. But I mean maybe EV_PERSIST is enough?